types/ui-select/index.d.ts

```diff
@@ -1,19 +1,77 @@
-// Type definitions for ui-select 0.13.2
+// Type definitions for ui-select 0.19.8
 // Project: https://github.com/angular-ui/ui-select
 // Definitions by: Niko Kovačič <https://github.com/nkovacic>
-// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped 
+//                 Adam Kwiatek <https://github.com/akwiatek>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.3
 
 /// <reference types="angular" />
 
 import * as angular from 'angular';
 
 declare module 'angular' {
-	export namespace ui.select {
-	    interface ISelectConfig {
-	        appendToBody: boolean;
-	        resetSearchInput: boolean;
-	        theme: string;
-	    }
-	}
+    export namespace ui.select {
+        interface ISelectConfig {
+            appendToBody: boolean;
+            backspaceReset: boolean;
+            closeOnSelect: boolean;
+            dropdownPosition: string;
+            generateId(): number;
+            paste?: IPasteFn;
```

Comment:
https://github.com/angular-ui/ui-select/blob/master/src/uiSelectController.js#L19 . Optional as it not defined in common.js